src/Symfony/Component/Serializer/Tests/Normalizer/ObjectNormalizerTest.php

```diff
@@ -953,6 +953,17 @@ public function testDenormalizeWithPropertyPath()
 
         $this->assertEquals($expected, $obj);
     }
+
+    public function testObjectNormalizerWithAttributeLoaderAndObjectHasStaticProperty()
+    {
+        $class = new class {
+            public static string $foo;
+        };
+
+        $normalizer = new ObjectNormalizer(new ClassMetadataFactory(new AttributeLoader()));
+        $normalizer->normalize($class);
+        $this->expectNotToPerformAssertions();
```

Comment:
There should be some assertion on what we expect as the result instead.

Reply:
Good point. I looked again and found `expectNotToPerformAssertions()`. I only had the methods `expectException*()` in mind, which assert when an exception is thrown. This is clearer.

```php
public function testObjectNormalizerWithAttributeLoaderAndObjectHasPublicStaticProperty()
{
    $this->expectNotToPerformAssertions();

    $class = new class {
        public static string $foo;
    };

    $normalizer = new ObjectNormalizer(new ClassMetadataFactory(new AttributeLoader()));
    $normalizer->normalize($class);
}
```

Reply:
I have adjusted the title and description of the PR. If the property is **public** and static, an exception is thrown. So nothing can be checked. Therefore, the result of normalise() cannot be tested.

Reply:
Then the test is missing a call of `expectException()`.

Reply:
_// I had deleted my comment._
I have adjusted the title and description of the PR. If the property is **public** and static, an exception is thrown. So nothing can be checked. Therefore the test that no exception is thrown.

> Then the test is missing a call of `expectException()`.

I have to correct myself. The result of `normalizer()` is an empty array if the exception is not thrown. For whatever reason. I have not checked this yet. 

No exception must be thrown. Due to the current state, the exception is thrown because `$foo` is considered cachable. This cache is being accessed, causing the exception. 

The question is whether the exception should be thrown at all? Is this the desired behaviour? I am not sure at the moment.

The condition was included in v6.4.9. In v7.0.0 the ObjectNormalizer was rebuilt/adapted. In v7.1.12 the condition was reintroduced as is.

Reply:
One more hint. The error occurs with the following ObjectNormalizer configuration:

```php
new ObjectNormalizer(new ClassMetadataFactory(new AttributeLoader()))
```

Reply:
> Then the test is missing a call of `expectException()`.

I misunderstood your comment. I thought I should test for an exception, but you wrote "missing a call of". 

The approach in PHPUnit would be to use try-catch, but I should remove that. I can reverse it.

`normalise()` returns an empty array, regardless of the type of the static property. Therefore, an `assertSame()` does not make an are, because it is not relevant either, but rather that the exception is not thrown. 

```php
try {
    $class = new class {
        public static string $foo;
    };
    
    $normalizer = new ObjectNormalizer(new ClassMetadataFactory(new AttributeLoader()));
    $normalizer->normalize($class);

    $this->assertTrue(true);
} catch (\Exception $e) {
    $this->fail($e->getMessage());
}
```

Reply:
```
$this->assertSame([], $normalizer->normalize($class));
```

IMO having such an assertion is useful since we will ensure this way that static properties are not part of the normalised result.

Reply:
I still have the title of the ticket in my head that it comes with an array the error message. But yes, static properties are not processed at all. So the array is always empty during the test.

The structure of the test should now be as follows?

```php
public function testObjectNormalizerWithAttributeLoaderAndObjectHasStaticProperty()
{
    $class = new class {
        public static string $foo;
    };

    $normalizer = new ObjectNormalizer(new ClassMetadataFactory(new AttributeLoader()));
    $this->assertSame([], $normalizer->normalize($class));
}
```